govcd/vm.go

```diff
@@ -247,44 +309,45 @@ func (vm *VM) ChangeCPUCountWithCore(virtualCpuCount int, coresPerSocket *int) (
 
 }
 
-func (vm *VM) ChangeNetworkConfig(networks []map[string]interface{}, ip string) (Task, error) {
+func (vm *VM) ChangeNetworkConfig(networks []map[string]interface{}) (Task, error) {
 	err := vm.Refresh()
 	if err != nil {
 		return Task{}, fmt.Errorf("error refreshing VM before running customization: %v", err)
 	}
 
 	networkSection, err := vm.GetNetworkConnectionSection()
 
-	// changes network config when only matches network name with provided one
-	for _, network := range networks {
-		for index, networkConnection := range networkSection.NetworkConnection {
-			if networkConnection.Network == network["orgnetwork"] { // network name are equal
+	for index, network := range networks {
+		for nicSlot, networkConnection := range networkSection.NetworkConnection {
+			// changes network config only if network name matches with provided one and if order is correct (this is needed
+			// for example if you attach NICs from same network)
+			if networkConnection.Network == network["orgnetwork"] && index == nicSlot {
 				// Determine what type of address is requested for the vApp
 				ipAllocationMode := "NONE"
 				ipAddress := "Any"
 
 				// TODO: Review current behaviour of using DHCP when left blank
-				if ip == "dhcp" || network["ip"].(string) == "dhcp" {
+				if network["ip"].(string) == "dhcp" {
```

Comment:
Can you doublecheck my testcode piece again. I still get some difference and 
https://play.golang.org/p/Wrf169V7Znd

Reply:
In my opinion this behavior was broken and now it's fixed. If you add an actual ip to your test you will see how this improves the code. ONLY if ip is valid and not dhcp allocated or none it will get to MANUAL otherwise just use DHCP as fallback
```
==== Before ====
dhcp 		Any 		DHCP
allocated 		Any 		POOL
none 		Any 		NONE
 		Any 		DHCP
RANDOM 		RANDOM 		MANUAL
127.0.0.1 		127.0.0.1 		MANUAL
==== After2 ====
dhcp 		Any 		DHCP
allocated 		Any 		POOL
none 		Any 		NONE
 		Any 		
RANDOM 		Any 		DHCP
127.0.0.1 		127.0.0.1 		MANUAL
```

Reply:
The only thing I'd consider changing is:
```
else {
		ipAllocationMode = network["ip_allocation_mode"].(string)
	}
```
could be:
```
else {
		ipAllocationMode = "DHCP"
	}
if network["ip_allocation_mode"].(string) {
		ipAllocationMode = network["ip_allocation_mode"].(string)
}
```